backend/queue-service/src/main/kotlin/com/ticketqueue/queue/scheduler/QueueCleanupScheduler.kt

```
@@ -0,0 +1,115 @@
+        logger.info { "Queue cleanup batch started" }
+        executedCounter.increment()
+
+        val scheduleIds = try {
+            eventServiceClient.getEndedScheduleIds().scheduleIds
+        } catch (e: RetryableException) {
+            logger.error(e) { "Queue cleanup batch failed: Event Service 5xx error (transient)" }
+            return
+        } catch (e: BusinessException) {
+            logger.error(e) { "Queue cleanup batch failed: Event Service 4xx error (non-transient), code=${e.errorCode}" }
+            return
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)